### PR TITLE
Load From Gallery Features

### PR DESCRIPTION
**Pull request by henrymxu**
- Open a video from gallery
- Process it using pose detection
- Does not include the adding angles and stuff toto workout session

### CODE REVIEW

File: VirtualTrainer/PoseDetectorHelper.swift [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    
+    init() {
+        
+    }
+    
+    func getResults() {
+        
+    }
+    
+    func detectPose(in image: VisionImage) -> [Pose] {
```
Comment (henrymxu) [inline VirtualTrainer/PoseDetectorHelper.swift:32]: processing for the poses is not done yet (we don't determine the joint angles or whatever)

File: VirtualTrainer/PreCameraViewController.swift [hunk 2]
```diff
@@ -31,12 +35,45 @@ class PreCameraViewController: UIViewController {
...
+                workoutType: workoutType,
+                cameraAngle: cameraAngle != nil
+            )
+          print("workout initialized \(vc.workoutSession?.workoutType)")
+        }
+    }
+    
+    func imagePickerController(_ picker: UIImagePickerController, didFinishPickingMediaWithInfo info: [UIImagePickerController.InfoKey : Any]) {
+        let videoURL = info[UIImagePickerController.InfoKey.mediaURL] as! URL
+        print(videoURL)
```
Comment (henrymxu) [inline VirtualTrainer/PreCameraViewController.swift:50]: open image gallery

File: VirtualTrainer/PreCameraViewController.swift [hunk 3]
```diff
@@ -31,12 +35,45 @@ class PreCameraViewController: UIViewController {
...
         }
     }
     
+    // MARK: Actions
+    
     @IBAction func startWorkout(_ sender: Any) {
     }
     
+    @IBAction func uploadRecording(_ sender: Any) {
+        imagePickerController.sourceType = .photoLibrary
```
Comment (henrymxu) [inline VirtualTrainer/PreCameraViewController.swift:69]: open image gallery

File: VirtualTrainer/PreCameraViewController.swift [hunk 4]
```diff
@@ -31,12 +35,45 @@ class PreCameraViewController: UIViewController {
...
+            vc.workoutSession = WorkoutSession(
+                workoutType: workoutType,
+                cameraAngle: cameraAngle != nil
+            )
+          print("workout initialized \(vc.workoutSession?.workoutType)")
+        }
+    }
+    
+    func imagePickerController(_ picker: UIImagePickerController, didFinishPickingMediaWithInfo info: [UIImagePickerController.InfoKey : Any]) {
+        let videoURL = info[UIImagePickerController.InfoKey.mediaURL] as! URL
```
Comment (henrymxu) [inline VirtualTrainer/PreCameraViewController.swift:49]: once user selected one, dismiss controller, get video frames, send to pose detector, then get results and start a segue to the feedback view

**Comment by n63li** [comment]
IMG_EA29AD5E1CC5-1
If we are going to upload videos in the `PreCameraViewController`, should we remove these two buttons? They don't serve any purpose then.

File: VirtualTrainer/PreCameraViewController.swift [hunk 5]
```diff
@@ -31,12 +35,45 @@ class PreCameraViewController: UIViewController {
                 cameraAngle: cameraAngle != nil
             )
           print("workout initialized \(vc.workoutSession?.workoutType)")
```
Comment (n63li) [inline VirtualTrainer/PreCameraViewController.swift:37]: My bad, can we remove these print statements? Left them in accidentally 😅

File: VirtualTrainer/PreCameraViewController.swift (hunk 2 above)
Comment (n63li) [inline VirtualTrainer/PreCameraViewController.swift:50]: Is this a debug statement?

**Comment by henrymxu** [comment]
> IMG_EA29AD5E1CC5-1
> If we are going to upload videos in the `PreCameraViewController`, should we remove these two buttons? They don't serve any purpose then.

yea i can remove that

**Review by n63li — APPROVED** [review]
LGTM 👍

**Pull request merged**